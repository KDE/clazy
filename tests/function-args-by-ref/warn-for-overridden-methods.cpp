struct NonTrivial {
    NonTrivial() {}
    NonTrivial(const NonTrivial &) {}
    void constFunction() const {};
    void nonConstFunction() {};
    int a;
};

class BaseWithVirtuals
{
public:
    virtual void virtualMethod1(const NonTrivial&) {}; // Warn
    virtual void virtualMethod2(const NonTrivial&) {}; // Warn
    void nonVirtualMethod(const NonTrivial&) {}; // Warn
};

class DerivedWithVirtuals : BaseWithVirtuals {
public:
    void virtualMethod1(const NonTrivial&) override {}; // Warn
    void virtualMethod2(const NonTrivial&) {}; // Warn
    void nonVirtualMethod(const NonTrivial&) {}; // Warn
};
