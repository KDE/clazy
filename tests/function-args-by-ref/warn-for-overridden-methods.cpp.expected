function-args-by-ref/warn-for-overridden-methods.cpp:12:33: warning: BaseWithVirtuals::virtualMethod1: Missing reference on non-trivial type (NonTrivial) [-Wclazy-function-args-by-ref]
function-args-by-ref/warn-for-overridden-methods.cpp:13:33: warning: BaseWithVirtuals::virtualMethod2: Missing reference on non-trivial type (NonTrivial) [-Wclazy-function-args-by-ref]
function-args-by-ref/warn-for-overridden-methods.cpp:14:27: warning: BaseWithVirtuals::nonVirtualMethod: Missing reference on non-trivial type (NonTrivial) [-Wclazy-function-args-by-ref]
function-args-by-ref/warn-for-overridden-methods.cpp:19:25: warning: DerivedWithVirtuals::virtualMethod1: Missing reference on non-trivial type (NonTrivial) [-Wclazy-function-args-by-ref]
function-args-by-ref/warn-for-overridden-methods.cpp:20:25: warning: DerivedWithVirtuals::virtualMethod2: Missing reference on non-trivial type (NonTrivial) [-Wclazy-function-args-by-ref]
function-args-by-ref/warn-for-overridden-methods.cpp:21:27: warning: DerivedWithVirtuals::nonVirtualMethod: Missing reference on non-trivial type (NonTrivial) [-Wclazy-function-args-by-ref]
